plugins/catalog-react/src/components/EntityKindPicker/kindFilterUtils.ts

```diff
@@ -14,48 +14,21 @@
  * limitations under the License.
  */
 
-import { useApi } from '@backstage/core-plugin-api';
-import useAsync from 'react-use/esm/useAsync';
-import { catalogApiRef } from '../../api';
-
-/**
- * Fetch and return all availible kinds.
- */
-export function useAllKinds(): {
-  loading: boolean;
-  error?: Error;
-  allKinds: string[];
-} {
-  const catalogApi = useApi(catalogApiRef);
-
-  const {
-    error,
-    loading,
-    value: allKinds,
-  } = useAsync(async () => {
-    const items = await catalogApi
-      .getEntityFacets({ facets: ['kind'] })
-      .then(response => response.facets.kind?.map(f => f.value).sort() || []);
-    return items;
-  }, [catalogApi]);
-
-  return { loading, error, allKinds: allKinds ?? [] };
-}
-
 /**
  * Filter and capitalize accessible kinds.
  */
 export function filterKinds(
```

Comment:
I wonder if it would be better to accept the `allowedKinds` and `forcedKinds` optional parameters in `useAllKinds()` instead and do the filtering there?